File: docs/snippets/go/config/keyword_config.go
package main

import (
	"fmt"
	"kreuzberg"
)

// Example 1: Basic YAKE configuration
// Uses YAKE algorithm with default parameters and English stopword filtering
func basicYake() error {
	config := &kreuzberg.ExtractionConfig{
		Keywords: &kreuzberg.KeywordConfig{
			Algorithm:   "yake",
			MaxKeywords: 10,
			MinScore:    0.0,
			NgramRange:  [2]int{1, 3},
			Language:    "en",
			YakeParams:  nil,
			RakeParams:  nil,
		},
	}

	result, err := kreuzberg.ExtractFile("document.pdf", "", config)
	if err != nil {
		return err
	}

	fmt.Printf("Keywords: %v\n", result.Keywords)
	return nil
}

// Example 2: Advanced YAKE with custom parameters
// Fine-tunes YAKE with custom window size for co-occurrence analysis
func advancedYake() error {
	config := &kreuzberg.ExtractionConfig{
		Keywords: &kreuzberg.KeywordConfig{
			Algorithm:   "yake",
			MaxKeywords: 15,
			MinScore:    0.1,
			NgramRange:  [2]int{1, 2},
			Language:    "en",
			YakeParams: &kreuzberg.YakeParams{
				WindowSize: 1,
			},
			RakeParams: nil,
		},
	}

	result, err := kreuzberg.ExtractFile("document.pdf", "", config)
	if err != nil {
		return err
	}

	fmt.Printf("Keywords: %v\n", result.Keywords)
	return nil
}

// Example 3: RAKE configuration
// Uses RAKE algorithm for rapid keyword extraction with phrase constraints
func rakeConfig() error {
	config := &kreuzberg.ExtractionConfig{
		Keywords: &kreuzberg.KeywordConfig{
			Algorithm:   "rake",
			MaxKeywords: 10,
			MinScore:    5.0,
			NgramRange:  [2]int{1, 3},
			Language:    "en",
			YakeParams:  nil,
			RakeParams: &kreuzberg.RakeParams{
				MinWordLength:      1,
				MaxWordsPerPhrase:  3,
			},
		},
	}

	result, err := kreuzberg.ExtractFile("document.pdf", "", config)
	if err != nil {
		return err
	}

	fmt.Printf("Keywords: %v\n", result.Keywords)
	return nil
}

func main() {
	if err := basicYake(); err != nil {
		fmt.Println("Error:", err)
	}
}

File: docs/snippets/ruby/config/keyword_config.rb
require 'kreuzberg'

# Example 1: Basic YAKE configuration
# Uses YAKE algorithm with default parameters and English stopword filtering
def basic_yake
  config = Kreuzberg::ExtractionConfig.new(
    keywords: Kreuzberg::KeywordConfig.new(
      algorithm: :yake,
      max_keywords: 10,
      min_score: 0.0,
      ngram_range: [1, 3],
      language: 'en',
      yake_params: nil,
      rake_params: nil
    )
  )

  result = Kreuzberg.extract_file('document.pdf', config: config)
  puts "Keywords: #{result.keywords}"
end

# Example 2: Advanced YAKE with custom parameters
# Fine-tunes YAKE with custom window size for co-occurrence analysis
def advanced_yake
  config = Kreuzberg::ExtractionConfig.new(
    keywords: Kreuzberg::KeywordConfig.new(
      algorithm: :yake,
      max_keywords: 15,
      min_score: 0.1,
      ngram_range: [1, 2],
      language: 'en',
      yake_params: Kreuzberg::YakeParams.new(
        window_size: 1
      ),
      rake_params: nil
    )
  )

  result = Kreuzberg.extract_file('document.pdf', config: config)
  puts "Keywords: #{result.keywords}"
end

# Example 3: RAKE configuration
# Uses RAKE algorithm for rapid keyword extraction with phrase constraints
def rake_config
  config = Kreuzberg::ExtractionConfig.new(
    keywords: Kreuzberg::KeywordConfig.new(
      algorithm: :rake,
      max_keywords: 10,
      min_score: 5.0,
      ngram_range: [1, 3],
      language: 'en',
      yake_params: nil,
      rake_params: Kreuzberg::RakeParams.new(
        min_word_length: 1,
        max_words_per_phrase: 3
      )
    )
  )

  result = Kreuzberg.extract_file('document.pdf', config: config)
  puts "Keywords: #{result.keywords}"
end

basic_yake if __FILE__ == $0
